### DISCUSSION TITLE
Qryn is dropping my tracing information which was generated by opentelementry

### DISCUSSION THREAD

**Discussion by yatamraghu7** [metrico/gigapipe#575] [Q&A] [answered]
Hi All,

I can see that qryn is dropping the tracing information like traceId, spanId which was generated for my request flow and I cannot find these information in qryn dashboard. I am using Http line instead of grpc and below are the screenshots.

My spring Boot Logs:

![Screenshot from 2024-09-16 17-19-40](https://github.com/user-attachments/assets/6dc95e04-b33c-4e31-83d4-26237b328773)

My Qryn Dashboard:

![Screenshot from 2024-09-16 17-20-43](https://github.com/user-attachments/assets/69fd3c24-d727-4756-8c93-a7e61b6ba491)

It would be great if someone can look into our code here 

https://github.com/yatamraghu7/ManualInstrumentationOtel

Thanks

**Comment by akvlad** [comment]
It's very unlikely for qryn to drop something. Are you sure you set the right labels for the fields?

Please try checking for the raw data inside the qryn database. Like:

```
SELECT * FROM samples_v3 
WHERE string LIKE '%dd2355bd377%' AND timestamp_ns >= <TIMESTAMP FOR 2024-09-16 17:09:41 IN THE TIMEZONE OF YOUR SCREENSHOT>
```
And check the result.

**Reply by yatamraghu7** [reply]
Hi @akvlad 

Thanks for the command. I tried as below and getting command not found. We are using clickhouse database. Can you please suggest if something needs to be changed.

![Screenshot from 2024-09-17 09-21-27](https://github.com/user-attachments/assets/d30bc538-0848-4ae7-9b75-cf80aee3bfde)

Thanks

**Reply by yatamraghu7** [reply]
Hi @akvlad 

I can see that in my logs that was produced in console contains the traceId and spanId and when it comes to qryn I cannot see that. Below are the screenshots

![Screenshot from 2024-09-17 09-54-27](https://github.com/user-attachments/assets/5a9fd261-b146-4d5f-8957-8e334f2c115d)

![Screenshot from 2024-09-17 09-55-49](https://github.com/user-attachments/assets/3bbeea80-ad5d-497a-aea3-2e65dcd53598)

I tried the command that was provided earlier and no luck. Can you help me in understanding where my tracing information was lost.

Thanks

**Reply by lmangani** [reply]
@yatamraghu7 you need to run this in clickhouse, not qryn

```
SELECT * FROM qryn.samples_v3 
WHERE string LIKE '%dd2355bd377%' AND timestamp_ns >= <TIMESTAMP FOR 2024-09-16 17:09:41 IN THE TIMEZONE OF YOUR SCREENSHOT>
````

**Reply by yatamraghu7** [reply]
Hi @lmangani 

I did ran the provided command inside the clickhouse only and not able to get any results

![Screenshot from 2024-09-17 10-25-36](https://github.com/user-attachments/assets/7d8df3e0-4094-48a8-90d8-65ff787e7fdd)

Thanks

**Reply by lmangani** [reply, marked as answer]
@yatamraghu7 you can't just run SQL in bash! You need to use a clickhouse-client or CLI to query the database: https://clickhouse.com/docs/en/interfaces/cli

Alternatively you can open port 8123 on the clickhouse-server container and use the ClickHouse PLAY UI to run the query

**Reply by yatamraghu7** [reply]
Hi @lmangani 

Thanks for pointing out the mistake that I was doing and now I have connected to clickhouse and ran the query got below error.

![Screenshot from 2024-09-17 12-26-29](https://github.com/user-attachments/assets/01cc0a4a-caaf-4f04-9943-fa0f913f5697)

Can you please suggest how can I move further.

Thanks

**Reply by yatamraghu7** [reply]
Hi @lmangani 

Thanks a lot for the support. I am now able to find the issue that when we are exporting the logs from our spring boot to opentelemetry collector which is running on 4318 port it is dropping our tracing information. So I will check with opentelemetry team.

Thanks

**Comment by yatamraghu7** [comment]
Thanks for the support
